WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/ReaderViewPage.java

```diff
@@ -0,0 +1,107 @@
+package org.wordpress.android.e2e.pages;
+
+import android.view.KeyEvent;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObject;
+import androidx.test.uiautomator.UiSelector;
+
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+import static org.wordpress.android.support.WPSupportUtils.DEFAULT_TIMEOUT;
+import static org.wordpress.android.support.WPSupportUtils.isTextDisplayed;
+import static org.wordpress.android.support.WPSupportUtils.swipeToLeft;
+import static org.wordpress.android.support.WPSupportUtils.swipeToRight;
+
+public class ReaderViewPage {
+    UiDevice mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+
+    String mLikerContainerId = "org.wordpress.android.prealpha:id/liker_faces_container";
+    String mRelatedPostsId = "org.wordpress.android.prealpha:id/container_related_posts";
+
+    UiObject mLikerContainer = mDevice.findObject(new UiSelector().resourceId(mLikerContainerId));
+    UiObject mRelatedPostsContainer = mDevice.findObject(new UiSelector().resourceId(mRelatedPostsId));
+    UiObject mSwipeForMore = mDevice.findObject(new UiSelector().textContains("Swipe for more"));
+
+    public ReaderViewPage waitUntilLoaded() {
+        mRelatedPostsContainer.waitForExists(DEFAULT_TIMEOUT);
+
+        return this;
+    }
+
+    public ReaderViewPage like() {
+        tapLikeButton();
+        mLikerContainer.waitForExists(DEFAULT_TIMEOUT);
+
+        return this;
+    }
+
+    public ReaderViewPage unlike() {
+        tapLikeButton();
+        mLikerContainer.waitUntilGone(DEFAULT_TIMEOUT);
+
+        return this;
+    }
```

Comment:
What do you think about adding a check prior to 'tapping'? This way we ensure we don't like an already liked post, and have no chance of waiting for absence / presence of an object that should never be absent / present.

Reply:
Yeah, I thought about that while I was writing the test and it felt like I'd be already testing before getting to the assertion. The `waitForExists` and `waitUntilGone` kind of does this as well but it also deals with the time it takes to load the liker avatar. Also, as our tests are all mocked, we always know the button previous state.

I tried it anyways in [21b34b2](https://github.com/wordpress-mobile/WordPress-Android/pull/15782/commits/21b34b2590bfe40feca062968e4f7632e6f43e9f) but got it reverted as it failed right away.

Reply:
> it felt like I'd be already testing before getting to the assertion

What I meant is a kind of service check (like checking if checkbox is already checked and returning, if this is the case, in a method that should set the checkbox to true). This way, it's not interfering into the test and asserting earlier - it's rather ensuring that the test does not try things it should not do. Maybe this approach comes from my experience with E2E tests and real API, when the initial state of elements theoretically could be not the expected one, and I'm not that used to the 'sterility' WireMock provides.

Reply:
Yeah, I got you. This check could also allow us to give a better error message in case the expected initial state is not met. However we would still have to wait for something to appear/disappear after the action, being it within likePost()/unlikePost() or in the assertion method. We can definitely revisit it.